ISHPullUp/ISHPullUpViewController.m

```diff
@@ -399,23 +400,23 @@ - (void)setBottomHidden:(BOOL)bottomHidden animated:(BOOL)animated {
         [bottomVC viewWillDisappear:animated];
     }
 
-    void (^updateBlock)();
-    updateBlock = ^{
+    void (^updateBlock)() = ^{
         [self setDimmingViewHidden:[self dimmingViewShouldBeHidden] height:self.bottomHeight];
         [self updateViewLayoutBottomHeight:self.bottomHeight withSize:self.view.bounds.size];
     };
 
-    void (^completion)(BOOL);
-    completion = ^(BOOL finished) {
-        if (!finished) {
+    __weak typeof(self) weakSelf = self;
+    void (^completion)(BOOL) = ^(BOOL finished) {
+        __strong typeof(self) strongSelf = weakSelf;
```

Comment:
No need for `typeof(self)`... just use `ISHPullUpViewController  *`. `__strong` is probably not needed either, as that's the default for local variables.